reana_client/cli/common.py

```diff
@@ -0,0 +1,125 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2018 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation; either version 2 of the License, or (at your option) any later
+# version.
+#
+# REANA is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# REANA; if not, write to the Free Software Foundation, Inc., 59 Temple Place,
+# Suite 330, Boston, MA 02111-1307, USA.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization or
+# submit itself to any jurisdiction.
+"""REANA client cli functionality shared by multiple cli commands."""
+
+import click
+import logging
+import os
+import traceback
+
+
+def get_current_click_command(ctx):
+    """Return command (and subcommands) that were used to create given context.
+
+    For example, a context that was created by a command
+    '<executable name> code upload', would return 'code.upload'.
+
+    """
+    command = []
+    while ctx.parent:
+        command.append(ctx.info_name)
+        ctx = ctx.parent
+    return '.'.join(str(x) for x in reversed(command))
+
+
+def upload_files_to_server(user, organization, workflow, filenames, client):
```

Comment:
Regarding having this as a `click` utility function, I still think that maybe not all the function but part of it should  be part of the Pythonic API the module offers, really soon CAP will ingest not only one but several file because the inputs/code for an analysis won't be just one or two files, what do you think, shall we have a quick talk about the design IRL?

Reply:
Just to double check the function signature would be
```
def upload_files_to_server(user, organization, workflow, filenames, api_client, upload_type):
    :param user:
    :type user: str
    :param organization:
    :type organization: str
    :param workflow:
    :type workflow: str
    :param filenames:
    :type filenames: str[]
    :param api_client:
    :type api_client: Bravado SwaggerClient
    :param upload_type:
    :type upload_type: str/enum either 'code' or 'upload'
```